### DISCUSSION TITLE
Supercollider File recognized as a Markdown file while there is another extension with .scd

### DISCUSSION THREAD

**Discussion by Xon77** [community/community#23511] [Projects and Issues]
<p>Hello,</p>
<p>For a project with SuperCollider, <a href="https://github.com/Xon77/Live4Life/blob/35f3f139ec4d618e3a89896cb8cbc59c061c4222/4Live4Life_Project/_0_Init_Live4Life.scd" rel="noopener nofollow ugc">this file</a> is recognized and colorised as as a Markdown file, while it is SuperCollider code and the extension is .scd; and I lose line numbers.</p>
<p>How can I get the proper colorisation and line numbers with this .scd file ?</p>
<p>I had to change the extension to another with .sc to get a more adequate view with line numbers with <a href="https://github.com/Xon77/Live4Life/blob/35f3f139ec4d618e3a89896cb8cbc59c061c4222/4Live4Life_Project/_0_Init_Live4Life.sc" rel="noopener nofollow ugc">this other file</a>. But the colorisation is still wrong because I should have an extension with .scd</p>
<p>Many thanks</p>
<p>Christophe</p>

**Comment by Xon77** [comment]
<p>I do not understand your answer.<br>
It has nothing to do with my question.</p>

**Comment by jsoref** [comment, marked as answer]
<p>GitHub file detection is handled by this repository:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/github/linguist">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/github/linguist" target="_blank" rel="noopener">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/2119212/181077480-2e6962dd-e3ea-4994-965f-03f673a7bca9.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/b/3ba8080b2672f3e6b0c4d89b682376d0072a72c2_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/b/3ba8080b2672f3e6b0c4d89b682376d0072a72c2_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/3/b/3ba8080b2672f3e6b0c4d89b682376d0072a72c2.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/b/3ba8080b2672f3e6b0c4d89b682376d0072a72c2_2_10x10.png"></div>

<h3><a href="https://github.com/github/linguist" target="_blank" rel="noopener">GitHub - github/linguist: Language Savant. If your repository's language is...</a></h3>

  <p>Language Savant. If your repository's language is being reported incorrectly, send us a pull request! - GitHub - github/linguist: Language Savant. If your repository's language is being rep...</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>They have documentation explaining the details and how to request changes.</p>
